homeassistant/components/subaru/sensor.py

```diff
@@ -146,140 +161,119 @@ async def async_setup_entry(
-            and self.hass.config.units == IMPERIAL_SYSTEM
-        ):
-            return round(
-                self.hass.config.units.pressure(self.current_value, self.api_unit),
+                unit_system.pressure(current_value, unit),
                 1,
             )
 
-        if (
-            self.api_unit in FUEL_CONSUMPTION_UNITS
-            and self.hass.config.units == IMPERIAL_SYSTEM
-        ):
-            return round((100.0 * L_PER_GAL) / (KM_PER_MI * self.current_value), 1)
+        if unit in FUEL_CONSUMPTION_UNITS and unit_system == IMPERIAL_SYSTEM:
+            return round((100.0 * L_PER_GAL) / (KM_PER_MI * current_value), 1)
 
-        return self.current_value
+        return current_value
 
     @property
     def native_unit_of_measurement(self):
         """Return the unit_of_measurement of the device."""
-        if self.api_unit in TEMPERATURE_UNITS:
-            return self.hass.config.units.temperature_unit
+        unit = self.entity_description.native_unit_of_measurement
 
-        if self.api_unit in LENGTH_UNITS:
+        if unit in LENGTH_UNITS:
             return self.hass.config.units.length_unit
 
-        if self.api_unit in PRESSURE_UNITS:
+        if unit in PRESSURE_UNITS:
             if self.hass.config.units == IMPERIAL_SYSTEM:
                 return self.hass.config.units.pressure_unit
-            return PRESSURE_HPA
 
-        if self.api_unit in FUEL_CONSUMPTION_UNITS:
+        if unit in FUEL_CONSUMPTION_UNITS:
             if self.hass.config.units == IMPERIAL_SYSTEM:
                 return FUEL_CONSUMPTION_MPG
-            return FUEL_CONSUMPTION_L_PER_100KM
 
-        return self.api_unit
+        return unit
 
     @property
     def available(self):
         """Return if entity is available."""
         last_update_success = super().available
         if last_update_success and self.vin not in self.coordinator.data:
             return False
+        if self.state is None:
+            return False
         return last_update_success
 
     def get_current_value(self):
         """Get raw value from the coordinator."""
-        value = self.coordinator.data[self.vin][VEHICLE_STATUS].get(self.data_field)
-        if value in sc.BAD_SENSOR_VALUES:
-            value = None
-        if isinstance(value, str):
-            if "." in value:
-                value = float(value)
-            else:
-                value = int(value)
-        return value
+        if isinstance(data := self.coordinator.data, dict):
+            value = data.get(self.vin)[VEHICLE_STATUS].get(self.entity_description.key)
+            if value in sc.BAD_SENSOR_VALUES:
+                value = None
+            if isinstance(value, str):
+                if "." in value:
+                    value = float(value)
+                elif value.isdigit():
+                    value = int(value)
+            _LOGGER.debug("Raw value for %s: %s", self._attr_name, value)
+            return value
```

Comment:
Side note: This kind of data parsing should really be done by the 3rd party library.

```suggestion
        if (data := self.coordinator.data) is None:
            return None
        value = data[self.vin][VEHICLE_STATUS].get(self.entity_description.key)
        if value in sc.BAD_SENSOR_VALUES:
            value = None
        if isinstance(value, str):
            if "." in value:
                value = float(value)
            elif value.isdigit():
                value = int(value)
        _LOGGER.debug("Raw value for %s: %s", self._attr_name, value)
        return value
```